### ISSUE TITLE
torch.utils.cpp_extension.CUDAExtension is still not ported

### ISSUE THREAD

**Issue by harish2704** [ROCmSoftwarePlatform/pytorch#382]
## 🐛 Bug

`torch.utils.cpp_extension` is the module used by many Pytorch extensions to build a  python module  
This module should support building a pytorch extension using ROCm libraries.

For example, there is a `CUDAExtension` class in this module. I wish if there were a similar class `HCCExtension` exists which enables us to build python extensions easly. 

## To Reproduce

I found this issue while trying to port `warp-ctc` library ( CTC loss function for Pytorch ) to ROCm. It is in WIP stage. Currently, I was able to compile the library, but couldn't build the python module.
WIP project repo: https://github.com/harish2704/warp-ctc

**Comment by iotamudelta** [comment]
HIPExtensions have landed upstream, torchvision is ROCm GPU kernel enabled.
